cloud/linode/loadbalancers.go

```diff
@@ -572,14 +660,77 @@ func (l *loadbalancers) getNodeBalancerByID(ctx context.Context, service *v1.Ser
 }
 
 func (l *loadbalancers) getLoadBalancerTags(_ context.Context, clusterName string, service *v1.Service) []string {
-	tags := []string{clusterName}
+	tags := []string{}
+	if clusterName != "" {
+		tags = append(tags, clusterName)
+	}
+
 	tagStr, ok := service.GetAnnotations()[annLinodeLoadBalancerTags]
 	if ok {
 		return append(tags, strings.Split(tagStr, ",")...)
 	}
 	return tags
 }
 
+func processACL(fwcreateOpts *linodego.FirewallCreateOptions, aclType, label, svcName, ports string, ips linodego.NetworkAddresses) {
+	fwcreateOpts.Rules.Inbound = append(fwcreateOpts.Rules.Inbound, linodego.FirewallRule{
+		Action:      aclType,
+		Label:       fmt.Sprintf("%s-%s", aclType, svcName),
+		Description: fmt.Sprintf("Created by linode-ccm: %s, for %s", label, svcName),
+		Protocol:    linodego.TCP, // Nodebalancers support only TCP.
+		Ports:       ports,
+		Addresses:   ips,
+	})
+	fwcreateOpts.Rules.OutboundPolicy = "ACCEPT"
+	if aclType == "ACCEPT" {
+		// if an allowlist is present, we drop everything else.
+		fwcreateOpts.Rules.InboundPolicy = "DROP"
+	} else {
+		// if a denylist is present, we accept everything else.
+		fwcreateOpts.Rules.InboundPolicy = "ACCEPT"
+	}
+}
+
+type aclConfig struct {
+	AllowList *linodego.NetworkAddresses `json:"allowList"`
+	DenyList  *linodego.NetworkAddresses `json:"denyList"`
+}
+
+func (l *loadbalancers) createFirewallOptsForSvc(label string, tags []string, svc *v1.Service) (*linodego.FirewallCreateOptions, error) {
+	// Fetch acl from annotation
+	aclString := svc.GetAnnotations()[annLinodeCloudFirewallACL]
+	fwcreateOpts := linodego.FirewallCreateOptions{
+		Label: label,
+		Tags:  tags,
+	}
+
+	servicePorts := make([]string, len(svc.Spec.Ports))
+	for idx, port := range svc.Spec.Ports {
+		servicePorts[idx] = strconv.Itoa(int(port.Port))
```

Comment:
Go antipattern, I'd rather `append`. The `servicePorts` definition will then have `len(svc.Spec.Ports)` as its capacity, not length.

Reply:
It wasn't happy about it, because you didn't do what I suggested :-)

You didn't initialise the slice (unlike now).

```
servicePorts := make([]string, 0, len(svc.Spec.Ports))
```

<img width="756" alt="image" src="https://github.com/linode/linode-cloud-controller-manager/assets/126059888/bfd1e49a-adeb-42e8-bb79-2f27c96bc0f6">